src/components/reference-book/page.cjsx

```diff
@@ -33,10 +40,47 @@ module.exports = React.createClass
       <i className='fa fa-chevron-right'/>
     </Router.Link>
 
+  hasTargetHash: ->
+    window.location.hash.length
+
+  # used by BookContentMixin
+  shouldOpenNewTab: -> false
+
+  getTargetEl: ->
+    targetSelector = window.location.hash
+    pageEl = @getDOMNode()
+    pageEl.querySelector(targetSelector)
+
+  scrollToTarget: (targetEl) ->
+    targetPosition = @getTopPosition(targetEl)
+    window.scrollTo(0, targetPosition)
+
+  triggerTargetHighlight: (targetEl) ->
+    targetEl.classList.add('target')
+    _.delay(->
+      targetEl.classList.remove('target')
+    , 1500)
+
+  componentDidMount: ->
+    return unless @hasTargetHash()
+
+    targetEl = @getTargetEl()
+    if targetEl?
+      @scrollToTarget(targetEl)
+      $imagesToLoad = $('img')
+      imagesLoaded = 0
+      $imagesToLoad.load(=>
```

Comment:
Can you add the load listener manually to each image (and remove jQuery)?

Something like:

```coffee
images = @getDOMNode().querySelectorAll('img')
imagesToLoad = images.length
onLoad = ->
  imagesToLoad -= 1
  if imagesToLoad is 0
    ...
for img in images
  img.addEventListener('load', onLoad)
```